addons/purchase_stock/tests/test_anglo_saxon_valuation_reconciliation.py

```diff
@@ -391,3 +391,52 @@ def test_reconcile_cash_basis_bill(self):
             {'debit': 0.0,     'credit': 50.0,      'amount_currency': -100.0,   'account_id': cash_basis_transfer_account.id},
             {'debit': 50.0,      'credit': 0.0,     'amount_currency': 100.0,  'account_id': tax_account_1.id},
         ])
+
+    def test_reconciliation_differed_billing(self):
+        """
+        Test whether products received billed at different time will be correctly reconciled
+        valuation: automated
+        - create a rfq
+        - receive products
+        - create bill - set quantity of product A = 0 - save
+        - create bill - confirm
+        -> the reconciliation should not take into account the lines of the first bill
+        """
+        date_po_and_delivery = '2022-03-02'
+        self.product_a.write({
+            'categ_id': self.stock_account_product_categ,
+            'detailed_type': 'product',
+            'uom_id': 1,
```

Comment:
Not that readable, I would prefer that you explicitly specify which UoM you want (same comment for `product_b`)

And in case of `product_a`, the UoM is already Unit:
https://github.com/odoo/odoo/blob/ea5347c082f58f5dbfb15214088ac68580e988fa/addons/account/tests/common.py#L79-L81

That being said, I'm not even sure it's relevant to update the UoMs ?